packages/client/hmi-client/src/workflow/ops/simulate-julia/tera-simulate-julia.vue

```diff
@@ -272,49 +225,37 @@ const onSelection = (id: string) => {
 const configurationChange = (index: number, config: ChartConfig) => {
 	const state = _.cloneDeep(props.node.state);
 	state.chartConfigs[index] = config.selectedVariable;
-
 	emit('update-state', state);
 };
 
 const addChart = () => {
 	const state = _.cloneDeep(props.node.state);
 	state.chartConfigs.push([]);
-
 	emit('update-state', state);
 };
 
-onMounted(() => {
-	const runIds = querySimulationInProgress(props.node);
-	if (runIds.length === 1) {
-		// there should only be one run happening at a time
-		getStatus(runIds[0]);
-	}
-});
-
-onUnmounted(() => {
-	poller.stop();
-});
-
 watch(
-	() => selectedRunId.value,
-	() => {
-		if (selectedRunId.value) {
-			lazyLoadSimulationData(selectedRunId.value);
-		}
-	},
-	{ immediate: true }
+	() => props.node.state.inProgressSimulationId,
+	(id) => {
+		if (id === '') showSpinner.value = false;
+		else showSpinner.value = true;
+	}
 );
```

Comment:
I think this could be done in a `computed` variable

Reply:
A watcher construct is better here, as it clearly signals the intention that the value being watched can and will change, whereas a computed just says it is dependent on other variables.

Also this construct allows you to do more things if needed.